src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -381,6 +381,41 @@
    "UPDATE factsets SET hash=md5(factsets.id::text)::bytea WHERE hash is NULL"
    "ALTER TABLE factsets ALTER COLUMN hash SET NOT NULL"))
 
+(defn add-support-for-historical-catalogs []
+  (jdbc/do-commands
+   "ALTER TABLE catalog_resources RENAME TO catalog_resources_tmp"
+   ;; CREATE certnames and catalog_resources transform tables
+   "CREATE TABLE catalog_resources (LIKE catalog_resources_tmp INCLUDING ALL)"
+   "CREATE TABLE latest_catalogs (catalog_id BIGINT REFERENCES catalogs(id) ON DELETE SET NULL, certname_id BIGINT PRIMARY KEY REFERENCES certnames(id) ON DELETE CASCADE)"
```

Comment:
Why create the primary key here if you're going to drop it and create a new one 2 lines below?